### PR TITLE
feat(text-area): add component tokens

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #7180 

## Summary
Add component tokens for calcite-text-area

Adds the following tokens:

``` 
  --calcite-text-area-background-color: Specifies the background color of the component.
  --calcite-text-area-border-block-end-color: Specifies the border block end color of the component.
  --calcite-text-area-border-color: Specifies the border color of the component.
  --calcite-text-area-character-limit-text-color: Specifies the color of the character limit text displayed in footer of the component.
  --calcite-text-area-text-color: Specifies the color of text in the component.
```

### CODE REVIEW

File: packages/calcite-components/src/components/text-area/text-area.scss [hunk 1]
```diff
@@ -1,66 +1,87 @@
...
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-text-area-background-color: Specifies the background color of the component.
+ * @prop --calcite--text-area-border-block-end-color: Specifies the border block end color of the component.
+ * @prop --calcite-text-area-border-color: Specifies the border color of the component.
+ * @prop --calcite-text-area-footer-min-block-size : Specifies minimum `block-size` of the footer in the component.
+ * @prop --calcite-text-area-min-inline-size: Specifies the mininmum `inline-size` of the component. Defaults to `192px` and increases to `288px` when `footer` is added.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/text-area/text-area.scss:12]: this should be an `-internal-` variable

File: packages/calcite-components/src/components/text-area/text-area.scss [hunk 2]
```diff
@@ -1,66 +1,87 @@
...
+
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-text-area-background-color: Specifies the background color of the component.
+ * @prop --calcite--text-area-border-block-end-color: Specifies the border block end color of the component.
+ * @prop --calcite-text-area-border-color: Specifies the border color of the component.
+ * @prop --calcite-text-area-footer-min-block-size : Specifies minimum `block-size` of the footer in the component.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/text-area/text-area.scss:11]: this should be an `-internal-` variable

File: packages/calcite-components/src/components/text-area/text-area.scss [hunk 3]
```diff
@@ -1,66 +1,87 @@
+@import "~@esri/calcite-design-tokens/dist/scss/core";
+
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-text-area-background-color: Specifies the background color of the component.
+ * @prop --calcite--text-area-border-block-end-color: Specifies the border block end color of the component.
```
Comment (alisonailea) [inline packages/calcite-components/src/components/text-area/text-area.scss:9]: change to `--calcite-text-area-border-color`

File: packages/calcite-components/src/components/text-area/text-area.scss [hunk 4]
```diff
@@ -76,94 +97,73 @@ textarea {
...
-  }
-  textarea {
-    @apply py-1
-    px-2;
-  }
-}
-
-:host([scale="s"]) {
-  textarea,
+  font-size: var(--calcite-font-size--2);
```
Comment (alisonailea) [inline packages/calcite-components/src/components/text-area/text-area.scss:100]: this is unnecssary in this PR

**Review by alisonailea — CHANGES_REQUESTED** [review]
Can this be broken into separate PRs and limit the theme-able component tokens to just the border, background, text, box-shadow

File: packages/calcite-components/src/components/text-area/text-area.scss (hunk 3 above)
Comment (anveshmekala) [reply packages/calcite-components/src/components/text-area/text-area.scss:9]: we have `--calcite-text-area-border-color` specified already. The `border-block-end-color` is different from the `border-color` for a valid text-area we need to define them separate.

File: packages/calcite-components/src/components/text-area/text-area.scss (hunk 4 above)
Comment (anveshmekala) [reply packages/calcite-components/src/components/text-area/text-area.scss]: reverted

**Review by alisonailea — APPROVED** [review]

File: packages/calcite-components/src/components/text-area/text-area.scss [hunk 5]
```diff
@@ -1,66 +1,86 @@
...
 :host {
   @apply inline-block
   relative
   w-full
   h-full;
+  --calcite-text-area-text-color: var(--calcite-color-text-1);
+  --calcite-text-area-background-color: var(--calcite-color-foreground-1);
+  --calcite-text-area-border-color: var(--calcite-color-border-input);
+  --calcite-text-area-border-block-end-color: var(--calcite-color-border-3);
+  --calcite-text-area-internal-min-inline-size: var(--calcite-size-192);
```
Comment (anveshmekala) [inline packages/calcite-components/src/components/text-area/text-area.scss:25]: This isn't working even with import of core. Thoughts? @alisonailea

Comment (anveshmekala) [reply packages/calcite-components/src/components/text-area/text-area.scss]: nvm, my bad.

**Pull request merged**